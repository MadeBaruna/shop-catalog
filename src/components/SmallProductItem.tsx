import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { Icon, Button } from 'semantic-ui-react';

const Div = styled.div`
  display: flex;
  flex-direction: column;
  background: white;
  padding: 8px;

  img {
    width: 100%;
    height: auto;
  }
`;

const DetailContainer = styled.div`
  display: flex;
  flex-direction: row;
  padding: 8px 0;
  margin-bottom: 8px;
`;

const Detail = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1;
  min-width: 0;

  p {
    font-size: 16px;
    margin-bottom: 4px;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    margin-right: 8px;
  }

  h3 {
    margin-top: 8px;
  }
`;

interface IProps {
  id: string;
  name: string;
  pictureUrl: string;
  size: string;
  price: number;
}

const priceFormat = new Intl.NumberFormat('id-ID');

const SmallProductItem: React.FC<IProps> = ({
  id,
  name,
  size,
  price,
  pictureUrl
}) => (
  <Link
    style={{ width: '50%' }}
    to={{
      pathname: `/product/${id}`,
      state: {
        title: name
      }
    }}
  >
    <Div>
      <img src={pictureUrl} alt="gambar produk" />
      <DetailContainer>
        <Detail>
          <p>{name}</p>
          <p>{priceFormat.format(price)}</p>
        </Detail>
        <Icon name="heart outline" color="grey" size="large" />
      </DetailContainer>
    </Div>
  </Link>
);

export default SmallProductItem;
