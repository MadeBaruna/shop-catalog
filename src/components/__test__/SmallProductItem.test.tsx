import React from 'react';

import SmallProductItem from '../SmallProductItem';
import { shallow } from 'enzyme';

it('renders <SmallProductItem />', () => {
  const wrapper = shallow(
    <SmallProductItem
      id="asd"
      name="This is a rather long product name"
      size="M, S, XL"
      pictureUrl="https://image.jpg"
      price={120000}
    />
  );

  expect(wrapper).toMatchSnapshot();
});
