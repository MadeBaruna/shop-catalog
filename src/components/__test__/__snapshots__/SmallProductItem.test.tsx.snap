// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <SmallProductItem /> 1`] = `
<Link
  style={
    Object {
      "width": "50%",
    }
  }
  to={
    Object {
      "pathname": "/product/asd",
      "state": Object {
        "title": "This is a rather long product name",
      },
    }
  }
>
  <styled.div>
    <img
      alt="gambar produk"
      src="https://image.jpg"
    />
    <styled.div>
      <styled.div>
        <p>
          This is a rather long product name
        </p>
        <p>
          120,000
        </p>
      </styled.div>
      <Icon
        as="i"
        color="grey"
        name="heart outline"
        size="large"
      />
    </styled.div>
  </styled.div>
</Link>
`;
